root/edit/details/EditRelationshipAttribute.js

```diff
@@ -0,0 +1,115 @@
+/*
+ * @flow
+ * Copyright (C) 2020 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import expand2react from '../../static/scripts/common/i18n/expand2react';
+import WordDiff from '../../static/scripts/edit/components/edit/WordDiff';
+
+type EditRelationshipAttributeEditT = {
+  ...EditT,
+  +display_data: {
+    +child_order?: CompT<number>,
+    +description: CompT<string | null>,
+    +name: CompT<string>,
+    +original_description: string | null,
+    +original_name: string,
+    +parent?: CompT<LinkAttrTypeT | null>,
+  },
+};
+
+type Props = {
+  +edit: EditRelationshipAttributeEditT,
+};
+
+const EditRelationshipAttribute = ({edit}: Props): React.Element<'table'> => {
+  const display = edit.display_data;
+  const childOrder = display.child_order;
+  const oldDescription = display.description.old;
+  const newDescription = display.description.new;
+  const descriptionRawChanges = newDescription !== oldDescription;
+  const newExpandedDescription = newDescription
+    ? expand2react(newDescription)
+    : '';
+  const oldExpandedDescription = oldDescription
+    ? expand2react(oldDescription)
+    : '';
+  const descriptionChanges =
+    newExpandedDescription !== oldExpandedDescription;
```

Comment:
If `expand2react` returns a React element (like say a fragment), then these will always be `!==`. They'd only be equal if they refer to the exact same object in memory. Or am I misunderstanding what this is doing?

In 83fd3a04a6782faed1725e7d315c1a96bfe0e0bb we started showing diffs of the raw HTML for link type edits, and I think it works well because the majority of the description edits seem to be small wording or tag adjustments and not rewrites. It'd be nice to keep some consistency between them in either case.

Reply:
Would you also show the HTML in AddRelationshipAttribute?

I'm thinking about always displaying both raw and final, but have only raw be a real diff. That way, we'd allow people to see the actual HTML change *and* how it will look like on the site.